cloudinit/net/freebsd.py

```diff
@@ -50,10 +51,8 @@ def start_services(self, run=False):
         for dhcp_interface in self.dhcp_interfaces():
             # Observed on DragonFlyBSD 6. If we use the "restart" parameter,
             # the routes are not recreated.
-            subp.subp(
-                ["service", "dhclient", "stop", dhcp_interface],
-                rcs=[0, 1],
-                capture=True,
+            net.dhcp.IscDhclient.start_service(
+                dhcp_interface, distros.freebsd.Distro
             )
```

Comment:
I don't think we can make this shift as-written. These definitions are no longer the same, the new `IscDhclient.start_service` method completely ignores the dhcp_interface because we are using `distro.manage_interface` and dropping the supplemental `dhcp_interface` argument in the refactor.

Instead of running:
`service dhclient stop eth0` 
we now run only the following on FreeBSD
`service dhclient stop`

Reply:
Nice catch, thanks. This should be fixed in [f1eb8ac](https://github.com/canonical/cloud-init/pull/2122/commits/f1eb8acc98b0995cfb7f00c0c24adfa7f951b604).